tests/systemtests/bank_test.go

```
@@ -0,0 +1,252 @@
+	assert.Equal(t, int64(initialAmount), balance2)
+
+	bankSendCmdArgs := []string{"tx", "bank", "send", account1Addr, account2Addr, "1000stake"}
+
+	testCases := []struct {
+		name         string
+		extraArgs    []string
+		expectErr    bool
+		expectedCode uint32
+	}{
+		{
+			"valid transaction",
+			[]string{"--fees=1stake"},
+			false,
+			0,
+		},
+		{
+			"not enough fees",
+			[]string{"--fees=0stake"},
+			true,
+			sdkerrors.ErrInsufficientFee.ABCICode(),
+		},
+		{
+			"not enough gas",
+			[]string{"--fees=1stake", "--gas=10"},
+			true,
+			sdkerrors.ErrOutOfGas.ABCICode(),
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			cmdArgs := append(bankSendCmdArgs, tc.extraArgs...)
+
+			if tc.expectErr {
+				assertErr := func(xt assert.TestingT, gotErr error, gotOutputs ...interface{}) bool {
+					assert.Len(t, gotOutputs, 1)
+					code := gjson.Get(gotOutputs[0].(string), "code")
+					assert.True(t, code.Exists())
+					assert.Equal(t, int64(tc.expectedCode), code.Int())
+					return false // always abort
+				}
+				rsp := cli.WithRunErrorMatcher(assertErr).Run(cmdArgs...)
+				RequireTxFailure(t, rsp)
+			} else {
+				rsp := cli.Run(cmdArgs...)
+				txResult, found := cli.AwaitTxCommitted(rsp)
+				assert.True(t, found)
+				RequireTxSuccess(t, txResult)
+			}
+		})
+	}
+
+	// test tx bank send with insufficient funds
+	insufficientCmdArgs := bankSendCmdArgs[0 : len(bankSendCmdArgs)-1]
+	insufficientCmdArgs = append(insufficientCmdArgs, initialBalance, "--fees=10stake")
+	rsp := cli.Run(insufficientCmdArgs...)
+	RequireTxFailure(t, rsp)
+	assert.Contains(t, rsp, sdkerrors.ErrInsufficientFunds.Error())
+
+	// test tx bank send with unauthorized signature
+	assertUnauthorizedErr := func(t assert.TestingT, gotErr error, gotOutputs ...interface{}) bool {
+		assert.Len(t, gotOutputs, 1)
+		code := gjson.Get(gotOutputs[0].(string), "code")
+		assert.True(t, code.Exists())
```

nit: here and others. prefer `require.` instead of `assert.` when you have a guard to fail fast.